src/core/service.c

```diff
@@ -214,6 +214,16 @@ static int service_set_main_pidref(Service *s, PidRef pidref_consume, const dual
 
         if (!pidref_equal(&s->main_pid, &pidref)) {
                 service_unwatch_main_pid(s);
+
+                dual_timestamp pid_start_time;
+
+                if (!start_timestamp) {
+                        usec_t t;
+
+                        if (pidref_get_start_time(&pidref, &t) >= 0)
+                                start_timestamp = dual_timestamp_from_realtime(&pid_start_time, t);
```

Comment:
so this isn't right afaics. should be dual_timestamp_from_boottime()

with this code here that assumes CLOCK_REALTIME you should get completely wrong times, no?